pkg/tsdb/cloudwatch/log_query_test.go

```diff
@@ -538,7 +538,126 @@ func TestGroupingResultsWithNumericField(t *testing.T) {
 		},
 	}
 
-	groupedResults, err := groupResults(fakeDataFrame, []string{"httpresponse"})
+	groupedResults, err := groupResults(fakeDataFrame, []string{"httpresponse"}, false)
+	require.NoError(t, err)
+	assert.ElementsMatch(t, expectedGroupedFrames, groupedResults)
+}
+
+func TestGroupingResultsWithRemoveNonNumericTrue(t *testing.T) {
```

Comment:
This test was a bit tough to digest. I have some suggestions, feel free to take em or leave em!
* I tried to remove anything more than the “minimum” in order for the test to be more concise — tried not to remove something you were testing. but I also feel like you can probably get rid of some fakelog-b and reduce this even more…
* simplified the time so there’s no noise to read about parsing the format
* inlined some variables — goal is for reader to consume as much as possible as a whole from top to bottom

```
func TestGroupingResultsWithRemoveNonNumericTrue(t *testing.T) {
	logField := data.NewField(“@log”, data.Labels{}, []*string{
		aws.String(“fakelog-a”),
		aws.String(“fakelog-b”),
		aws.String(“fakelog-a”),
		aws.String(“fakelog-b”),
	})

	countField := data.NewField(“count”, data.Labels{}, []*string{
		aws.String(“100"),
		aws.String(“34”),
		aws.String(“57"),
		aws.String(“200”),
	})

	timeA := time.Time{}
	timeB := time.Time{}.Add(1 * time.Minute)
	fakeDataFrame := &data.Frame{
		Name: “CloudWatchLogsResponse”,
		Fields: []*data.Field{
			data.NewField(“@timestamp”, data.Labels{}, []*time.Time{&timeA, &timeA, &timeB, &timeB}),
			logField,
			countField,
		},
	}

	expectedGroupedFrames := []*data.Frame{
		{
			Name: “fakelog-a”,
			Fields: []*data.Field{
				data.NewField(“@timestamp”, data.Labels{}, []*time.Time{
					&timeA, &timeB,
				}),
				data.NewField(“count”, data.Labels{}, []*string{
					aws.String(“100”),
					aws.String(“57"),
				}),
			},
		},
		{
			Name: “fakelog-b”,
			Fields: []*data.Field{
				data.NewField(“@timestamp”, data.Labels{}, []*time.Time{
					&timeA, &timeB,
				}),
				data.NewField(“count”, data.Labels{}, []*string{
					aws.String(“34"),
					aws.String(“200”),
				}),
			},
		},
	}

	groupedResults, err := groupResults(fakeDataFrame, []string{“@log”}, true)
	require.NoError(t, err)
	assert.ElementsMatch(t, expectedGroupedFrames, groupedResults)
}
```